notebooks/concepts/Fitting_multiprocess.ipynb

```diff
@@ -0,0 +1,1742 @@
+    "#     print(x, y, fitted_model_xy.amplitude_0)\n",
+    "        \n",
+    "t2 = time.time() - t1 \n",
+    "t3 = t2 / (2*HSIZE+1)**2\n",
+    "print(\"Total time: \", t2, \"s\")\n",
+    "print(\"Time per spaxel:\", t3, \"s\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, to check if calling the `Spectrum1D` constructor for every single spaxel may incur in a penalty, we run the same test case, but using a single `Spectrum1D` instance intitalized with the flux cube."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Total time:  98.26243805885315 s\n",
+      "Time per spaxel: 0.10225019569079412 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "# (code repeated here for clarity).\n",
+    "region = SpectralRegion(6500*wunit, 7000*wunit)\n",
+    "spectrum = Spectrum1D(spectral_axis=wave, flux=flux3d[:, CENTER, CENTER] * funit)\n",
+    "i1, i2 = _to_edge_pixel(region.subregions[0], spectrum)\n",
+    "\n",
+    "# Slice to extract region\n",
+    "wave_r = wave[i1:i2]\n",
+    "flux3d_r = flux3d[i1:i2,:,:]\n",
+    "\n",
+    "# Transpose the flux cube as per Spectrum1D requirement.\n",
+    "flux3d_r_transposed = flux3d_r.transpose(1,2,0)\n",
+    "\n",
+    "# Build the 3D Spectrum1D instance.\n",
+    "spectrum = Spectrum1D(spectral_axis=wave_r, flux=flux3d_r_transposed * funit)\n",
+    "\n",
+    "# Main timing loop\n",
+    "t1 = time.time() \n",
+    "for x,y in SPAXELS:\n",
+    "\n",
+    "    fitted_model_xy = fit_lines(spectrum[x,y], fitted_model_central)\n",
+    "        \n",
+    "#     print(x, y, fitted_model_xy.amplitude_0)\n",
+    "        \n",
+    "t2 = time.time() - t1\n",
+    "t3 = t2 / (2*HSIZE+1)**2\n",
+    "print(\"Total time: \", t2, \"s\")\n",
+    "print(\"Time per spaxel:\", t3, \"s\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Conclusion"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There seems to be no penalty in calling the `Spectrum1D` constructor for every spaxel. The difference in execution times between the two approaches is of a few percent for cubes with up to 2500 spaxels.\n",
+    "\n",
+    "This is actually important to know, for the reason that follows:\n",
+    "\n",
+    "It hapens that passing a `Spectrum1D` instance to a multiprocess pool worker, prevents the worker from being called by the pool. The spectrum instance does not have even to participate in the actual execution, just a reference to it, if stored in the worker instance, somehow inhibits the worker from being called by the pool.\n",
+    "\n",
+    "This was tested with the worker object as an instance of a class, as below, and also with the worker being a plain top level function to which the spectrum gets passed at every call. No difference. "
+   ]
```

Comment:
So you can't pass a Spectrum1D to a pool worker?  I ran into a similar problem with passing an lmfit method to multiprocessing pool.  My solution was to wrap it in a higher-level function.

Reply:
I tried several different methods for passing a spectrum to the pool, but nothing worked. The only way that it worked for me is the one in the notebook, namely, pass a bare array and instantiate a Spectrum1D in the worker function.  Fortunately that doesn't seem to be a costly solution in terms of execution speed. I propose we do the implementation in this way and tackle the pool mysterious behavior in a separate issue.